src/adt/internedstateset.c

```diff
@@ -9,1235 +9,522 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <string.h>
+#include <stdio.h>
 
 #include <fsm/fsm.h>
 
 #include <adt/alloc.h>
 #include <adt/stateset.h>
-#include <adt/internedstateset.h>
 
 #include "common/check.h"
 
-/* Adapted from _Confluently Persistent Sets and Maps_ by Olle Liljenzin.
- *
- * This is a confluent persistent set pool, for interning sets of state
- * IDs. If multiple sets are constructed with the same elements, they
- * will end up with references to the same root node no matter what
- * order the IDs are added, which allows constant-time testing for
- * equality. Partially overlapping sets will be built using shared
- * subtrees when possible, which can save quite a lot of memory.
- *
- * A hash table is used to intern individual nodes, and reference
- * counting tracks when nodes can be returned to the freelist and
- * reused. There can be a lot of short-lived nodes constructed during
- * subtree rebalancing. Reference counting keeps the working set small,
- * but generational tracing collection may also be worth exploring at
- * some point. */
```

Comment:
I think an equivalent orientation comment would still be helpful, even if building the iss incrementally no longer applies